openmdao/api.py

```diff
@@ -40,10 +39,8 @@
 from openmdao.solvers.run_once import RunOnce
 from openmdao.solvers.scipy_gmres import ScipyGMRES
 from openmdao.solvers.solver_base import LinearSolver, NonLinearSolver
-try:
+if under_mpirun():
     from openmdao.solvers.petsc_ksp import PetscKSP
```

Comment:
Maybe there's something we could do on the system side, so MPI4py doesn't crash so spectacularly. I take it importing MPI4py outside MPI works on your cluster?